### DISCUSSION TITLE
Access the relationship using `Select` Field

### DISCUSSION THREAD

**Discussion by mahmoudmohamedramadan** [orchidsoftware/platform#1990] [Q&A] [answered]
When I was dealing WITH the `Select` Field I needed to access the `posts` relationship of the currently authenticated user BUT without avail. Any help?

**Comment by tabuna** [comment]
Form an accessible list in the query method;

```php
public function query(): array
{
    $this->personalOptionsForTheUser = [
        'key'   => 'value',
    ];
}
```

After that, you can pass it through the property:

```php
Select::make('select')
    ->options($this->personalOptionsForTheUser)
```

If the field is used in another class, you can use `$this->query->get('key')` to access the screen data. You can read about this here: https://orchid.software/en/docs/rows/#accessing-screen-data

**Reply by mahmoudmohamedramadan** [reply]
Thanks :)

**Comment by mahmoudmohamedramadan** [comment, marked as answer]
I wrote this code and I got the correct result

```PHP
Select::make('user.post')
                ->fromModel(\App\Models\Post::class, 'title')
                ->required()
                ->title(__('Title'))
                ->help('Specify which title you should belong to'),
```
